ompi/mca/io/romio314/src/io_romio314_component.c

```diff
@@ -234,15 +235,29 @@ static int delete_query(const char *filename, struct ompi_info_t *info,
 }
 
 
-static int delete_select(const char *filename, struct ompi_info_t *info,
+static int delete_select(const char *filename, struct opal_info_t *info,
                          struct mca_io_base_delete_t *private_data)
 {
     int ret;
 
+// Technically an opal_info_t isn't a full ompi_info_t. so if we're
+// using an MPI call below with an MPI_Info, we need to create an equivalent
+// MPI_Info:
+// In the current design a lot of code is written to use opal_info_t.
+// I'd be open to reverting most of it back to ompi_info_t and using
+// opal_info_* calls on the opal_info_t that's inside the ompi_info_t.
+// I don't think there's any advantage to most of the code being switched
+// to opal_info_t, which is causing the unnecessary extra complexity here:
```

Comment:
Seems like this comment should be addressed...? (a similar comment is repeated several times)